### ISSUE TITLE
[IDEA/SUGGESTION] Cache gTTS samples and reuse them

### ISSUE THREAD

**Issue by MegaBrutal** [wootguy/Radio#1]
I've been thinking about the flooding problem (when players flood TTS with the same word and you get a ban from GoogleTranslate). Reuse already spoken phrases by building a cache of them.

If I read it correctly, here you save the samples to files anyway:
https://github.com/wootguy/Radio/blob/d1702d8ca378b510c847a9e263ce949a8c9ad60d/radio_client.py#L294

When players repeat the same phrase, you could use the already saved sample. Of course, this would imply that you save files with a different naming convention, and it would only be useful for relatively short, but common phrases, so I'm not really sure if it would worth the effort. Also it wouldn't stop players from simply flooding the system with different phrases. Rest assured, some rate limiting would probably help.

I'd love to provide a pull request, but I don't (yet?) have a test environment for Sven Co-op server configs. I only have a HLDM server so far – as you might guess, that's easier to maintain. :D

**Comment by wootguy** [comment]
yeah that's a good idea. I've got like 100GB free space on the VPS that's runing this so saving everything would probably be no big deal.

Today was the first time it got banned, and it most likely happened because plugins were disabled and people were flooding the chat for a full minute. Normally the anti-spam plugin would have prevented that.

I need get some instructions set up for this too. It's kind of complicated to set up. Lots of config stuff is hard-coded, too.
